x/evm/keeper/state_transition.go

```diff
@@ -343,10 +343,12 @@ func (k *Keeper) ApplyMessageWithConfig(msg core.Message, tracer vm.Tracer, comm
 	}
 
 	if contractCreation {
+		// revert nonce increment in ante handler, because it'll be increased again in evm.
 		nonce := k.GetNonce(sender.Address())
+		if nonce > 0 {
```

Comment:
this shouldn't happen afaik, right? I'd suggest switching to a `panic`

```go
		if nonce == 0 {
				panic("nonce cannot be 0 ...")	
		}
```

Reply:
I got a better idea, just `k.SetNonce(sender.Address(), msg.Nonce())` here, and we don't have to change the test cases and grpc queries.